pkg/api/org_invite.go

```diff
@@ -143,6 +145,9 @@ func GetInviteInfoByCode(c *m.ReqContext) Response {
 	}
 
 	invite := query.Result
+	if invite.Status != m.TmpUserInvitePending {
+		return Error(404, "Invite not found", nil)
```

Comment:
Maybe better use `412` status to conform with [this](https://github.com/grafana/grafana/blob/master/pkg/api/org_invite.go#L167).

Reply:
I don't really know how [CompleteInvite](https://github.com/grafana/grafana/blob/master/pkg/api/org_invite.go#L155) is invoked in a user flow, so it's difficult for me to evaluate how these two API endpoints should correlate with each other.

In isolation, 404 seems more useful to me for this particular endpoint, since I don't think it should matter to the user which state the invite is in on the back-end if it shouldn't be accessed.

Reply:
I agree 404 is more logical than 412. When submitting invite-form rendered at invite url the CompleteInvite is the handler of that action.

Reply:
I think that the CompleteInvite is [called](https://github.com/grafana/grafana/blob/master/public/app/core/controllers/invited_ctrl.ts#L36) by the frontend when one submits the signup (after invitation form) but I don't have a strong opinion about this either.